hal/src/boron/network/sara_ncp_client.cpp

```diff
@@ -1271,4 +1336,26 @@ int SaraNcpClient::modemSetUartState(bool state) const {
     return 0;
 }
 
+void SaraNcpClient::waitForPowerOff() {
+    LOG(TRACE, "Modem waiting up to 30s to power off with PWR_UC...");
+    system_tick_t now = millis();
+    if (powerOnTime_ == 0) {
+        powerOnTime_ = now; // fallback to max timeout of 30s to be safe
+    }
+    do { // check for timeout (VINT == LOW, Powered on 30s ago, Registered 20s ago)
+        now = millis();
+        // prefer to timeout 20s after registration if we are registered
+        if (registeredTime_) {
+            if (now - registeredTime_ >= 20000UL) {
+                break;
+            }
+        } else if (powerOnTime_ && now - powerOnTime_ >= 30000UL) {
+            break;
+        }
+        HAL_Delay_Milliseconds(1); // just wait
```

Comment:
Let's increase this to something like `100`.

Reply:
Ok, done.